src/main/java/net/snowflake/client/core/SqlInputTimestampUtil.java

```diff
@@ -0,0 +1,45 @@
+package net.snowflake.client.core;
+
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.util.TimeZone;
+import net.snowflake.client.jdbc.SnowflakeUtil;
+import net.snowflake.common.core.SnowflakeDateTimeFormat;
+
+public class SqlInputTimestampUtil {
+
+  @SnowflakeJdbcInternalApi
+  public static Timestamp getTimestampFromType(
+      int columnSubType,
+      String value,
+      SFBaseSession session,
+      TimeZone sessionTimeZone,
+      TimeZone tz) {
+    if (columnSubType == SnowflakeUtil.EXTRA_TYPES_TIMESTAMP_LTZ) {
+      return getTimestampFromFormat(
+          "TIMESTAMP_LTZ_OUTPUT_FORMAT", value, session, sessionTimeZone, tz);
+    } else if (columnSubType == SnowflakeUtil.EXTRA_TYPES_TIMESTAMP_NTZ
+        || columnSubType == Types.TIMESTAMP) {
+      return getTimestampFromFormat(
+          "TIMESTAMP_NTZ_OUTPUT_FORMAT", value, session, sessionTimeZone, tz);
+    } else if (columnSubType == SnowflakeUtil.EXTRA_TYPES_TIMESTAMP_TZ) {
+      return getTimestampFromFormat(
+          "TIMESTAMP_TZ_OUTPUT_FORMAT", value, session, sessionTimeZone, tz);
+    } else {
+      return null;
```

Comment:
It is correct to return here null or maybe throw exception or some default?

Reply:
I would leave this null value here, it is being handled when this method is called. When it returns null -> we use our custom converters.